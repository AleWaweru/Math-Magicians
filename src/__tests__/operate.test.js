import operate from '../logic/operate';

describe('operate', () => {
  test('adds two numbers', () => {
    expect(operate(2, 3, '+')).toMatchSnapshot();
  });

  test('subtracts two numbers', () => {
    expect(operate(5, 2, '-')).toMatchSnapshot();
  });

  test('multiplies two numbers', () => {
    expect(operate(3, 4, 'x')).toMatchSnapshot();
  });

  test('divides two numbers', () => {
    expect(operate(10, 2, 'รท')).toMatchSnapshot();
  });

  test('returns an error message if dividing by 0', () => {
    expect(operate(5, 0, 'รท')).toMatchSnapshot();
  });

  test('calculates the modulo of two numbers', () => {
    expect(operate(5, 2, '%')).toMatchSnapshot();
  });

  test('returns an error message if finding modulo when dividing by 0', () => {
    expect(operate(5, 0, '%')).toMatchSnapshot();
  });

  test('throws an error for unknown operations', () => {
    expect(() => operate(2, 3, '*')).toThrowErrorMatchingSnapshot();
  });
});
