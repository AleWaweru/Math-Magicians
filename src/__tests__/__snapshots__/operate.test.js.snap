// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operate adds two numbers 1`] = `"5"`;

exports[`operate calculates the modulo of two numbers 1`] = `"1"`;

exports[`operate divides two numbers 1`] = `"5"`;

exports[`operate multiplies two numbers 1`] = `"12"`;

exports[`operate returns an error message if dividing by 0 1`] = `"Can't divide by 0."`;

exports[`operate returns an error message if finding modulo when dividing by 0 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`operate subtracts two numbers 1`] = `"3"`;

exports[`operate throws an error for unknown operations 1`] = `"Unknown operation '*'"`;
